### PR TITLE
Adds the Proton C base project

### PR DESCRIPTION
**Pull request by jackhumbert**

### CODE REVIEW

File: keyboards/proton_c/rules.mk [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# Options to pass to dfu-util when flashing
+DFU_ARGS = -d 0483:df11 -a 0 -s 0x08000000:leave
+
+# Build Options
+#   comment out to disable the options.
+#
+BACKLIGHT_ENABLE = no
+BOOTMAGIC_ENABLE = yes  # Virtual DIP switch configuration
+## (Note that for BOOTMAGIC on Teensy LC you have to use a custom .ld script.)
```
Comment (drashna) [inline keyboards/proton_c/rules.mk]: Is this line relevant?

Comment (jackhumbert) [reply keyboards/proton_c/rules.mk]: No :/ we have a lot of comments in the code like this, and they often get out-dated quickly (not that this one is).

**Comment by drashna** [comment]
Also, I opened this like 30 seconds after you opened the PR. :D

File: keyboards/proton_c/readme.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Keyboard Maintainer: [Jack Humbert](https://github.com/jackhumbert)  
+Hardware Supported: QMK Proton C  
+Hardware Availability: [List of vendors](https://qmk.fm/proton-c)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make proton_c:default
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/proton_c/readme.md]: Please:

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

**Comment by mechmerlin** [comment]
Is this going to pop up on the configurator?

**Comment by drashna** [comment]
It would, most likely.   But I think it's there for easy reference for those wanting to make a board (since it's not as straightforward.   

Also, I think there was some codechange to make it easy convert over, eg a switch in the configurator aht would include all the Proton C stuff, and change the relevant settings. 

Though thinking about it... @jackhumbert shouldn't we add the board to "/quantum/template/" so that people could easily create new projects with the Proton C?

**Comment by jackhumbert** [comment]
For now, I think we should just exclude it until we have more of the options exposed.

A template might be nice, but I also want something that they can compile to actually test. I think an actual project in keyboards/ that people can copy from will be good.

File: keyboards/proton_c/config.h [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+#define DEVICE_VER      0x0006
+#define MANUFACTURER    QMK
+#define PRODUCT         Proton C
+#define DESCRIPTION     A compact ortholinear keyboard
+
+/* key matrix size */
+#define MATRIX_ROWS 1
+#define MATRIX_COLS 1
+
+#define MATRIX_ROW_PINS { C2 }
```
[the remark this answers is not in the document]
Comment (jackhumbert) [reply keyboards/proton_c/config.h]: Updated :)

**Pull request merged**